src/lib/command_line.ml

```diff
@@ -80,10 +81,25 @@ module Server = struct
     in
     Log.log t.log ~section:["server"; subsection] json
 
-  let save_job_list t =
-    Storage.Json.save_jsonable t.storage
-      ~path:["server"; "jobs.json"]
-      (`List (List.map t.jobs ~f:(fun j -> `String (Job.id j))))
+  let change_job_list t action =
+    let mutex =
+      match t.job_list_mutex with
+      | Some s -> s
+      | None ->
+        let m = Lwt_mutex.create () in
```

Comment:
`Lwt` is cooperative so there cannot be a preemption between the beginning of the function and the call to `Lwt_mutex.with_lock`.

The mutex is there to make sure updates `t.jobs <- ...` happen together with the `Storage.save_jsonable`.

I had to convince myself again ☺, so maybe I'll try to come up with something more readable/obvious.